instrumentation/fragment/src/main/java/io/opentelemetry/android/instrumentation/fragment/FragmentLifecycleInstrumentation.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.android.instrumentation.fragment
+
+import android.app.Application
+import android.app.Application.ActivityLifecycleCallbacks
+import android.os.Build
+import io.opentelemetry.android.OpenTelemetryRum
+import io.opentelemetry.android.instrumentation.AndroidInstrumentation
+import io.opentelemetry.android.instrumentation.common.Constants.INSTRUMENTATION_SCOPE
+import io.opentelemetry.android.instrumentation.common.ScreenNameExtractor
+import io.opentelemetry.android.internal.services.ServiceManager
+import io.opentelemetry.android.internal.services.visiblescreen.fragments.RumFragmentActivityRegisterer
+import io.opentelemetry.api.trace.Tracer
+
+class FragmentLifecycleInstrumentation : AndroidInstrumentation {
+    private var screenNameExtractor = ScreenNameExtractor.DEFAULT
+    private var tracerCustomizer: (Tracer) -> Tracer = { it }
+
+    fun setTracerCustomizer(customizer: (Tracer) -> Tracer) {
+        tracerCustomizer = customizer
+    }
+
+    fun setScreenNameExtractor(screenNameExtractor: ScreenNameExtractor) {
+        this.screenNameExtractor = screenNameExtractor
+    }
+
+    override fun install(
+        application: Application,
+        openTelemetryRum: OpenTelemetryRum,
+    ) {
+        application.registerActivityLifecycleCallbacks(buildFragmentRegisterer(openTelemetryRum))
+    }
+
+    private fun buildFragmentRegisterer(openTelemetryRum: OpenTelemetryRum): ActivityLifecycleCallbacks {
```

Comment:
This code was moved from [here](https://github.com/open-telemetry/opentelemetry-android/pull/465/files#diff-3699b73481a1f4e4e9f99c7361324d8395de5d012b7c5b8c0da0afac1e969b4bL91).